app/src/main/java/org/simple/clinic/selectcountry/SelectCountryUpdate.kt

```diff
@@ -31,10 +31,11 @@ class SelectCountryUpdate : Update<SelectCountryModel, SelectCountryEvent, Selec
   }
 
   private fun manifestFetched(model: SelectCountryModel, countries: List<Country>): Next<SelectCountryModel, SelectCountryEffect> {
+    val manifestFetchedModel = model.manifestFetched(countries)
     return if (countries.size > 1) {
-      next(model.manifestFetched(countries))
+      next(manifestFetchedModel)
     } else {
-      dispatch(SaveCountryEffect(countries.first()))
+      next(manifestFetchedModel, SaveCountryEffect(countries.first()))
```

Comment:
Just curious, do we need to update the model at all? Can't we just perform the side effect and done with it? Even if the saving country takes a bit more time (highly unlikely) we just progress indicator instead of blank screen. WDYT? 🤔

Reply:
Yeah, we can do that. I updated the model so `SelectCountryInit` does not fetch manifest again and again as it depends on `val effects: Set<SelectCountryEffect> = if (!model.hasFetchedCountries()) setOf(FetchManifest) else emptySet()`. Also the screen will only be rendered when the count is greater than 1 i.e. `model.hasFetchedCountries() && model.hasMoreThanOneCountry -> {
        ui.displaySupportedCountries(model.countries!!, model.selectedCountry)
      }`